.github/workflows/golangci-lint.yaml

```diff
@@ -0,0 +1,50 @@
+name: golangci-lint
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+      - main
+  pull_request:
+    paths:
+      - "src/**"
+      - "main.go"
+      - "go.mod"
+      - "go.sum"
+      - ".github/workflows/golangci-lint.yaml"
+permissions:
+  contents: read
+  # Optional: allow read access to pull request. Use with `only-new-issues` option.
+  # pull-requests: read
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v2
```

Comment:
this is kinda redundant if `golangci/golangci-lint-action@v2` is used.

Reply:
took it directly from their example but let's see if it can work without it (I'd be glad to drop it if it works)